os-eap-txn-partition/added/queryapi/query.py

```diff
@@ -0,0 +1,157 @@
+"""
+Copyright 2018 Red Hat, Inc.
+
+Red Hat licenses this file to you under the Apache License, version
+2.0 (the "License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied.  See the License for the specific language governing
+permissions and limitations under the License.
+"""
+
+import argparse
+import json
+import logging
+import urllib2
+
+from enum import Enum
+
+
+class QueryType(Enum):
+    """
+    Represents what could be queried.
+    PODS: list of pods
+    LOG: log from particular pod
+    """
+
+    PODS = 'pods'
+    PODS_LIVING = 'pods_living'
+    LOG = 'log'
+
+    def __str__(self):
+        return self.value
+
+class OutputFormat(Enum):
+    """
+    Represents output format of this script.
+    RAW: no formatting
+    LIST_SPACE: if possible values are delimited with space and returned
+    LIST_COMMA: comma separated list
+    """
+
+    RAW = "raw"
+    LIST_SPACE = "list_space"
+    LIST_COMMA = "list_comma"
+
+    def __str__(self):
+        return self.value
+
+
+class OpenShiftQuery():
+    """
+    Utility class to help query OpenShift api. Declares constant
+    to get token and uri of the query. Having methods doing the query etc.
+    """
+
+    API_URL = 'https://openshift.default.svc'
+    TOKEN_FILE_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/token'
+    NAMESPACE_FILE_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/namespace'
+    CERT_FILE_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt'
+    STATUS_LIVING_PODS = ['Pending', 'Running', 'Unknown']
```

Comment:
Do we need to be concerned with pods that might be bouncing, e.g. crash loop backoff?  (I'm not sure what state those would be in, or if there would be anything to recover if that happened.  Just asking.)

Reply:
yes, I think we need to be concerned but I got the reference for the pod possible status values that could be received from the API from the kubernetes doc: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-and-container-status
I could be wrong but I hope this is ok.

Reply:
Sounds good.  Just checking.